### PR TITLE
[WIP] Refactor the main Inspector panel (not its subpanels)

### PR DESCRIPTION
**Pull request by hbenl**
TODO:

- [x] reactivate the computed view
- [x] reactivate the searchbox
- [x] initialise the subpanels in the sidebar to the currently selected node
- [ ] remove old code

### CODE REVIEW

File: src/devtools/client/inspector/reducers.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  switch (action.type) {
+    case "set_inspector_3_pane_mode":
+      let activeTab = state.activeTab === "newruleview" ? "layoutview" : state.activeTab;
+      return { ...state, is3PaneModeEnabled: action.is3PaneModeEnabled, activeTab };
+    case "set_active_inspector_tab":
+      return { ...state, activeTab: action.activeTab };
+    default:
+      return state;
+  }
+}
```
Comment (jasonLaster) [inline src/devtools/client/inspector/reducers.ts:36]: minor detail, but in the other panels  we do `reducers/indexjs` 

we also have a UI reducer in `reducer/ui` which is what this new `inspector` reducer looks like

**Review by jasonLaster — COMMENTED** [review]
Wow!

**Review by jasonLaster — APPROVED** [review]
Works great

File: src/devtools/client/inspector/components/App.tsx [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+        activePanel = (
+          <div key={panel.id} className="tab-panel-box" role="tabpanel">
+            {panel.panel}
+          </div>
+        );
+      }
+
+      let className = "tabs-menu-item";
+      if (isPanelSelected) {
+        className += " is-active";
```
Comment (gabrielluong) [inline src/devtools/client/inspector/components/App.tsx:162]: We have a `classnames` util that might simplify this.

**Review by gabrielluong — APPROVED** [review]

**Pull request merged**